src/main/java/core/basesyntax/Lottery.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax;
+
+public class Lottery {
+    private Ball ball;
+    private ColorSupplier color;
+    private NumberBall numberBall;
```

Comment:
Think about which variables should be local in the method and which should be class-level. Here, 'ball' should not be a class-level field since a new Ball instance should be created and returned each time 'getRandomBall' is called.

Reply:
add     private Ball ball;